pkg/ext-proc/backend/pod_controller.go

```diff
@@ -0,0 +1,49 @@
+package backend
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+)
+
+type PodController struct {
+	client.Client
+	Scheme    *runtime.Scheme
+	Record    record.EventRecorder
+	Datastore *K8sDatastore
+}
+
+func (c *PodController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
We need to check health status.

I actually suggest that the LLMServerPool reconciler to create a Service, which will create EndpoinSlices, which will include the health/readiness status.

With this, we don't need a pod reconciler; we will need an EndpointSlice reconciler to update the data store with the healthy pods only. wdyt?

Reply:
The other option is to keep this pod reconciler, but check the readiness status of the pod remove unhealthy pods from the store

Reply:
I think I'll keep this PR as a pod reconciler. We can migrate to EndpointSlice later if we want, but I believe the only gain would be the management of the status/health of pods for us. Right?

Reply:
Discussed offline, working with EndpointSlices offers more info on the endpoints in one place, including health status, locality information etc.